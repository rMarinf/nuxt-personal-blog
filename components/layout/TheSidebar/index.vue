<template>
  <div v-click-outside="closeMobileMenu" class="side-menu">
    <a
      href="#"
      class="side-menu-nav-toggle"
      :class="{ active: isActive }"
      @click="showMobileMenu"
    >
      <i></i>
    </a>
    <aside
      id="side-menu-aside"
      class="img"
      :style="{
        backgroundImage: 'url(' + backgroundImageUrl + ')'
      }"
    >
      <the-sidebar-navigation></the-sidebar-navigation>
      <the-sidebar-footer></the-sidebar-footer>
    </aside>
  </div>
</template>

<script>
import ClickOutside from 'vue-click-outside'
import backgroundImageUrl from '~/assets/images/sidebar-bg.jpg'
import TheSidebarFooter from '@/components/layout/TheSidebar/components/TheSidebarFooter'
import TheSidebarNavigation from '@/components/layout/TheSidebar/components/TheSidebarNavigation'

export default {
  name: 'TheSidebar',
  components: { TheSidebarNavigation, TheSidebarFooter },
  directives: {
    ClickOutside
  },
  data() {
    return {
      isActive: false,
      backgroundImageUrl
    }
  },
  methods: {
    showMobileMenu(event) {
      event.preventDefault()
      const body = document.body
      if (body.classList.contains('offcanvas')) {
        this.isActive = false
        body.classList.remove('offcanvas')
      } else {
        this.isActive = true
        body.classList.add('offcanvas')
      }
    },
    closeMobileMenu() {
      const body = document.body
      if (body.classList.contains('offcanvas')) {
        body.classList.remove('offcanvas')
        this.isActive = false
      }
    }
  }
}
</script>

<style scoped></style>
