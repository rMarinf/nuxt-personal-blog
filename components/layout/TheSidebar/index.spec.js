import { shallowMount } from '@vue/test-utils'
import TheSidebar from './index.vue'

describe('TheSidebar', function() {
  describe('Render', function() {
    test('should render component', () => {
      // render the component
      const wrapper = shallowMount(TheSidebar, {
        stubs: {
          'the-sidebar-navigation': true,
          'the-sidebar-footer': true
        }
      })

      expect(wrapper.element).toMatchSnapshot()
    })

    test('should render component with active class', () => {
      // render the component
      const wrapper = shallowMount(TheSidebar, {
        stubs: {
          'the-sidebar-navigation': true,
          'the-sidebar-footer': true
        },
        data() {
          return {
            isActive: 'true'
          }
        }
      })

      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
