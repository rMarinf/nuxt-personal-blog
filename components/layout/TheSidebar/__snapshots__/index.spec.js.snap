// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheSidebar Render should render component 1`] = `
<div
  class="side-menu"
>
  <a
    class="side-menu-nav-toggle"
    href="#"
  >
    <i />
  </a>
   
  <aside
    class="img"
    id="side-menu-aside"
    style="background-image: url();"
  >
    <the-sidebar-navigation-stub />
     
    <the-sidebar-footer-stub />
  </aside>
</div>
`;

exports[`TheSidebar Render should render component with active class 1`] = `
<div
  class="side-menu"
>
  <a
    class="side-menu-nav-toggle active"
    href="#"
  >
    <i />
  </a>
   
  <aside
    class="img"
    id="side-menu-aside"
    style="background-image: url();"
  >
    <the-sidebar-navigation-stub />
     
    <the-sidebar-footer-stub />
  </aside>
</div>
`;
